shell/platform/cpp/BUILD.gn

```diff
@@ -0,0 +1,77 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("$flutter_root/common/config.gni")
+
+_common_defines = [ "FLUTTER_DESKTOP_LIBRARY" ]
+
+_public_headers = [
+  "public/flutter_export.h",
+  "public/flutter_messenger.h",
+  "public/flutter_plugin_registrar.h",
+]
+if (use_glfw_desktop) {
+  _public_headers += [ "public/flutter_glfw.h" ]
+}
+
+# The headers are a separate source set since the client wrapper is allowed
+# to depend on the public headers, but none of the rest of the code.
+source_set("common_cpp_library_headers") {
+  public = _public_headers
+
+  defines = _common_defines
+}
+
+source_set("common_cpp") {
+  public = [
```

Comment:
Ditto about the use of configs. Use a `public` and `private` config and targets can selectively import. For the SDK export, you can use a variable that both that copy target and the configs reference.